manuscripts2/new_functions.py

```diff
@@ -0,0 +1,510 @@
+import logging
+
+from collections import OrderedDict, defaultdict
+from datetime import timezone
+from dateutil import parser
+
+import pandas as pd
+from elasticsearch import Elasticsearch
+from elasticsearch_dsl import A, Q, Search
+
+logger = logging.getLogger(__name__)
+
+
+class Index():
+    """Index class which will represent an index in elasticsearch"""
+
+    def __init__(self, index, url=None):
```

Comment:
Instead of using an url, use an Elasticsearch object. This allows you to have all the power of an ES object, and don't need to re-create that functionality in this module. The pattern for using this Index object would be:

```
es = Elasticsearch(...)
index = Index(index_name, es)
```

If you need a complex Elasticsearch object, you just initialize it as needed (eg, a SSL instance without checking certificates, or whatever you may need).

I agree that default could be the "default" Elasticsearch instance, so that if you pass no es argument, you just produce in the initialization of the object a default Elasticsearch object, just by instatiating it as `Elasticsearch()` (afaik, by default it will use http://localhost:9200 ).